### ISSUE TITLE
Update website/documentation to use Bazel [phab:D265]

### ISSUE THREAD

**Issue by kythe** [google/kythe#482]
_Created by **schroederc** at 2015-05-19 22:01:13_:

**Comment by kythe** [comment]
_**schroederc** wrote at 2015-05-19 22:29:52_:

Remove more refs to campfire

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2015-05-19 23:21:07_:

8?

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2015-05-19 23:21:07_:

update.sh

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2015-05-19 23:21:07_:

8 (for Bazel, at least?)

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2015-05-19 23:21:07_:

Weren't we using KYTHE_INDEX_PACK for this elsewhere?


```
grep -Re 'KYTHE_INDEX_PACK' kythe
kythe/java/com/google/devtools/kythe/extractors/java/standalone/AbstractJavacWrapper.java: *   KYTHE_INDEX_PACK: if set to a non-empty value, interpret KYTHE_OUTPUT_DIRECTORY as the root of
kythe/java/com/google/devtools/kythe/extractors/java/standalone/AbstractJavacWrapper.java:        if (Strings.isNullOrEmpty(System.getenv("KYTHE_INDEX_PACK"))) {
kythe/docs/example-linux.txt:export KYTHE_INDEX_PACK=1
kythe/cxx/extractor/testdata/test_stdin_names.sh:echo '#define STDIN_OK 1\n' | KYTHE_INDEX_PACK=1 \
kythe/cxx/extractor/testdata/test_index_pack.sh:KYTHE_OUTPUT_DIRECTORY="${OUT_DIR}" KYTHE_INDEX_PACK="1" \
kythe/cxx/extractor/testdata/test_index_pack.sh:KYTHE_OUTPUT_DIRECTORY="${OUT_DIR}" KYTHE_INDEX_PACK="1" \
kythe/cxx/extractor/cxx_extractor_main.cc:// If KYTHE_INDEX_PACK is set to "1", the extractor will treat
kythe/cxx/extractor/cxx_extractor.cc:  if (const char* env_index_pack = getenv("KYTHE_INDEX_PACK")) {
```

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2015-05-19 23:21:07_:

as simple as:

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2015-05-19 23:21:07_:

Maybe note that 3.0.2 also works since this is the default version you get; our official supported version remains 2.3, though.

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2015-05-19 23:21:07_:

Maybe note that setup_bazel.sh should be run before update.sh.

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2015-05-19 23:21:07_:

Maybe remove "its build system, "

**Comment by kythe** [comment]
_**schroederc** wrote at 2015-05-20 16:37:31_:

review changes

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-05-20 16:37:36_:

That's a variable used by the script itself.  See #27

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-05-20 16:37:36_:

Done

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-05-20 16:37:36_:

Done

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-05-20 16:37:36_:

Sure

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-05-20 16:37:36_:

Done

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-05-20 16:37:36_:

Sure

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-05-20 16:37:36_:

Done

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-05-20 16:37:36_:

✓

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2015-05-20 16:49:57_:

shahms noted that this should probably be lxc-docker?

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2015-05-20 16:58:32_:

lxc-docker, which unfortunately requires a PPA.  docker.io is unmaintained and apparently contains various security vulnerabilities.

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2015-05-20 16:58:32_:

Do we actually need to install this ancient version of bison still?

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-05-20 16:59:29_:

It depends on the distro.  Debian uses 'docker.io'.  https://docs.docker.com/installation/debian/#debian-jessie-80-64-bit

**Comment by kythe** [comment]
_**schroederc** wrote at 2015-05-20 17:02:14_:

Link to Docker installation instructions

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-05-20 17:05:01_:

zarko, do we?

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2015-05-20 17:19:18_:

Fair enough, at some point it may be worthwhile to automate this on a per-distribution basis.

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2015-05-20 17:19:18_:

I assume we're planning on actually having something there eventually?  Why change the docs to refer to `/opt/kythe` rather than using `bazel run`

**Comment by kythe** [comment]
_**zarko** commented inline (context lost during migration) at 2015-05-20 17:22:53_:

3.0.2 works for me; this is the bison you get when you ask apt for it on trusty. We can probably drop our requirement for 2.3.

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-05-20 17:36:59_:

I want to push a v0.0.1 release this week so that something is there.

Building Kythe can be a pain.  I'd like to lessen the effort for entry so I'm making most of the public documentation use the release binaries.  Everything else is basically under the contributing page for now.

**Comment by kythe** [comment]
_**schroederc** wrote at 2015-05-20 17:40:55_:

Update bison version

**Comment by kythe** [comment]
_**schroederc** commented inline (context lost during migration) at 2015-05-20 17:41:59_:

I'm dropping the requirement but leaving 2.3 in the build containers for CI.

**Comment by kythe** [comment]
_**shahms** commented inline (context lost during migration) at 2015-05-20 17:45:05_:

That seems right.  I'm adding a comment to the grammar itself mentioning that it must remain compatible with bison 2.3

**Comment by kythe** [comment]
_**shahms** accepted this revision at 2015-05-20 18:08:30_

**Comment by kythe** [comment]
_**schroederc** closed via commit c9eff93f5a9f721c36aaa3f3921b428ad38afe8b at 2015-05-20 22:29:50_
